docs/manual/cluster-admin/how-to-customize-alerts.md

```diff
@@ -0,0 +1,144 @@
+# How to Customize Alerts
+
+OpenPAI supports the customization of alert rules and corresponding handling actions.
+The alert rules are managed by `prometheus` service and the handling actions are managed by `alert-manager` service.
+
+In this document, we will introduce existing alerts & actions, their matching methods, and how to add new customized alerts & actions.
+
+## Existing Alerts/Actions & How to Match Them 
+
+### Existing Alerts
+
+OpenPAI uses `Prometheus` to monitor system metrics.
+We provide various alerts by defining rules on virtual_clusters, GPU utilization, etc.
+If OpenPAI is deployed, you can then visit `your_master_ip/prometheus/alerts` to see the details of alerts, including their definition and status.
+
+For example, `PAIJobGpuPercentLowerThan0_3For1h` alert rule is by default configured as :
+
+``` yaml
+alert: PAIJobGpuPercentLowerThan0_3For1h
+expr: avg(task_gpu_percent{virtual_cluster=~"default"}) BY (job_name) < 0.3
+for: 1h
+```
+
+It means that the job on virtual cluster `default` with task level average GPU percent lower than `30%` for more than `1 hour` will fire the alert `PAIJobGpuPercentLowerThan0_3For1h`.
+
+### Existing Actions
+
+OpenPAI uses `alert-manager` service for alert handling. So far, we have provided three types of actions: 
+
+* webportal-alert: Show alerts on the home page of webportal.
+* email-alert: Send email to the assigned receiver.
+* webhook-alert: Send alert via a POST request, we now support stopping jobs by adapting the POST request to the corresponding REST API of OpenPAI.
+
+These actions can be configured in the `alert-manager` field of `service-configuration.yml`: 
+
+```yaml
+alert-manager:
+  webprotal-alert: 
+    enable: True
+  email-alert:
+    enable: False
+    receiver: your_addr@example.com
+    smtp_url: smtp.office365.com:587
+    smtp_from: alert_sender@example.com
+    smtp_auth_username: alert_sender@example.com
+    smtp_auth_password: password_for_alert_sender
+    port: 9093 # optional, do not change this if you do not want to change the port alert-manager is listening on
+  webhook-alert:
+    enable: False
+    port: 9095 # optional, do not change this if you do not want to change the port alert-handler is listening on
+    bearer_token: 'your_application_token_for_rest_server'
+```
+
+By default, `webportal-alert` is enabled, `email-alert` and `webhook-alert` are disabled.
+
+### How to Match Alerts and Actions
+
+The matching rules of alerts and actions are defined in [alert-configmap.yaml.template](https://github.com/microsoft/pai/blob/master/src/alert-manager/deploy/alert-configmap.yaml.template).
+
+
+If you enable `email-alert` and `webhook-alert` in `service-configuration.yml`, after rendering, this file will be like:
+
```

Comment:
actually I think you'd better give the example of template file other than the rendered result, because the template file is where admin needs to modify, right?
```markdown
The matching rules of alerts and actions are defined in [alert-configmap.yaml.template](https://github.com/microsoft/pai/blob/master/src/alert-manager/deploy/alert-configmap.yaml.template).

Here is the example of enabling `email-notification` and `stopping-job` actions for alert `PAIJobGpuPercentLowerThan0_3For1h`
```yaml
    route:
      receiver: pai-alert
      group_wait: 30s
      group_interval: 5m
      repeat_interval: {{ cluster_cfg["alert-manager"]["repeat-interval"] }}
      group_by: [alertname, alertstate]
      routes:
      - receiver: 'pai-alert-handler'
        match: 
          alertname: PaiJobLowGpuPercent
    
    receivers:
    - name: "pai-alert"
      email_configs:
        - to: {{ alert_info["receiver"] }}
          send_resolved: true
          html: '{{ "{{" }} template "email.pai.html" . {{ "}}" }}'
          headers:
            subject: '{{ cluster_cfg["cluster"]["common"]["cluster-id"] }}: {{ "{{" }} template "__subject" . {{ "}}" }}'
    - name: "pai-alert-handler"
      email_configs:
        - to: {{ alert_info["receiver"] }}
          send_resolved: true
          html: '{{ "{{" }} template "email.pai.html" . {{ "}}" }}'
          headers:
            subject: '{{ cluster_cfg["cluster"]["common"]["cluster-id"] }}: {{ "{{" }} template "__subject" . {{ "}}" }}'
      webhook_configs:
        - url: 'http://localhost:{{ alert_info["alert-handler"]["port"] }}/alert-handler'
          http_config:
            bearer_token: {{ alert_info["alert-handler"]["bearer_token"] }}
{% endif %}
```

```